### ISSUE TITLE
Creating tenants with Global environment variable does not save info in the appsettings.json of the tenant

### ISSUE THREAD

**Issue by jptissot** [OrchardCMS/OrchardCore#7597]
I have these environment set in a powershell console

```
$env:OrchardCore__ConnectionString="server=localhost;uid=root;database=db;password=my-secret-pw;"
$env:OrchardCore__DatabaseProvider="MySql"
```
If I setup a site / tenants while these are set, the ConnectionString and DatabaseProvider do not get output in the appsettings.json file of the tenant.

```json
{
  "Description": "test2",
  "RecipeName": "bootstrap-theme-setup",
  "Secret": "077908bc-f98b-45ce-a4cf-5b6fd4c60bc5",
  "TablePrefix": "test2222"
}
```
if I create the same site without the environment variables set, this get's output:
```json
{
  "ConnectionString": "server=localhost;uid=root;database=db;password=my-secret-pw;",
  "DatabaseProvider": "MySql",
  "Description": "test",
  "RecipeName": "bootstrap-theme-setup",
  "Secret": "af146286-5f82-46f9-a821-be92a5a57427",
  "TablePrefix": "test"
}
```

If I stop and run the site without the environment variables set, the `test2` site does not work.

/cc @jtkech

**Comment by jtkech** [comment]
Without saying it is okay in this case, yes it works as intended

Config source data are not mutable by the app, you need another tool (e.g. any editor to change an appsettings), one exception is our apsettings under the tenant folder that we can mutate through the admin acting as a config tool.

Config sources have different precedences, by default we keep the precedences of regular config sources, then our custom config sources have higher priorities. If a config value is already provided by a source, we may display it or not when editing a tenant, but anyway we persist it only if we change the already provided value.

This allow to change a config value depending on the context, e.g. through specific `appsettings.[env].json` files, or as here with different env variables values. Otherwise, if the value is always persisted e.g. in the appsettings under the tenant folder, because it has an higher priority, we would not be able to change the configuration according to a given context.

Use case: Having a different connection string in a dev environment for a tenant that is already setup.

But i understand your use case, maybe we need a different behavior for the special case of env variables (more ephemeral by nature), hmm maybe a env variable with a different name meaning that we want to persist the provided value.

**Comment by jptissot** [comment]
I understand what you are saying. And maybe it's ok that it works like this. I What I found strange is that I was seeing the ConnectionString and DatabaseProvider prefilled in the tenant setup page. But these values were not persisted, simply because they were coming from an environment variable. I would have assume that we should take a snapshot of these values when creating a tenant and write them in the appsettings.json file. 

I did not test this, but if I wonder what happens if I change the value of the connection string. If it is persisted in that case.
